### PR TITLE
VR-6030 Move log_model() to _DeployableEntity

### PR DESCRIPTION
**Pull request by nhatsmrt**
@convoliution I can't change the branch in the PR, so I just created a new one 🤷

### CODE REVIEW

**Review by conradoverta — APPROVED** [review]
LGTM

**Comment by convoliution** [comment]
Ooh boy, this is an old one I never got around to testing.

@nhatsmrt Could you run this branch in our python client test pipeline (after resolving conflicts 🙇), and make sure there's no `log_model()` related errors?

**Comment by nhatsmrt** [comment]
@convoliution No worries, I was going to do it anyway. The conflict resolution might take awhile tho, since I don't rmb the details of the PR.

**Comment by nhatsmrt** [comment]
Conflicts resolved. That took less time than I thought.

File: client/verta/verta/_tracking/deployable_entity.py [hunk 1]
```diff
@@ -429,6 +429,8 @@ def log_model(self, model, custom_modules=None, model_api=None, artifacts=None,
         except (TypeError, ValueError):
             extension = None
         _utils.THREAD_LOCALS.active_experiment_run = self
+        if isinstance(model, six.string_types):  # filepath
```
Comment (nhatsmrt) [inline client/verta/verta/_tracking/deployable_entity.py:435]: This is from Evgenii's recent torch PR.

File: client/verta/tests/test_cli/test_registry.py [hunk 2]
```diff
@@ -251,7 +252,8 @@ def test_create_version_with_custom_modules(self, client, registered_model, crea
                 "type": "torch",
                 "python_version": "2.7.17"
             }
-            model_version.log_artifact("model_api.json", model_api, True, "json")
+            self._log_artifact("model_api.json", model_api, _CommonCommonService.ArtifactTypeEnum.BLOB, 'json',
+                               overwrite=True)
```
Comment (nhatsmrt) [inline client/verta/tests/test_cli/test_registry.py]: This reminds me that we don't have any mechanism to log model_api for CLI 🤦

Comment (convoliution) [reply client/verta/tests/test_cli/test_registry.py]: I've filed VR-7696 to track this.

**Comment by convoliution** [comment]
Hey @nhatsmrt have you gotten a chance to test this PR out?

**Comment by nhatsmrt** [comment]
> Hey @nhatsmrt have you gotten a chance to test this PR out?

@convoliution Yes, but there were a few deployment bugs polluting the view. One is the comment you saw above, and the other was fixed in https://github.com/VertaAI/modeldb/pull/1621. I'll re-run the tests again. I hope there's no rush 😬

**Comment by convoliution** [comment]
@nhatsmrt Fun! Yep, no rush and no pressure—just passingly curious about the status, is all

**Comment by liuverta** [comment]
Oh dear, this has quite a few merge conflicts now.

And anyway, with https://github.com/VertaAI/modeldb/pull/3159, there should be a much simpler way to do this in the near-future.

**Pull request closed without merging**